### PR TITLE
[Imperative] implement imperative NCCLParallelContext

### PR DESCRIPTION
**Pull request by Yancey0623**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_imperative_nccl_context.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def test_nccl_context(self):
+        strategy = parallel.ParallelStrategy()
+
+        p_env = parallel.Env()
+        strategy.nranks = p_env.nranks
+        strategy.local_rank = p_env.local_rank
+        strategy.current_endpoint = p_env.current_endpoint
+        strategy.trainer_endpoints = p_env.trainer_endpoints
+
+        place = fluid.CUDAPlace(parallel.Env().dev_id)
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_nccl_context.py]: what if not gpu

File: paddle/fluid/imperative/nccl_context.cc [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/imperative/nccl_context.h"
+
+namespace paddle {
+namespace imperative {
+void NCCLParallelContext::RecvNCCLID(const std::string &ep,
```
Comment (panyx0718) [inline paddle/fluid/imperative/nccl_context.cc:20]: might want to test this and methods below

File: python/paddle/fluid/tests/unittests/test_imperative_nccl_context.py (hunk 1 above)
Comment (Yancey0623) [reply python/paddle/fluid/tests/unittests/test_imperative_nccl_context.py]: Will import Gloo to sync the var, and will implement the GlooParallelContext in the future.

File: paddle/fluid/imperative/nccl_context.cc (hunk 2 above)
Comment (Yancey0623) [reply paddle/fluid/imperative/nccl_context.cc:20]: done.

**Review by panyx0718 — DISMISSED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**